constellations/gemini-user/src/achievements/achievements.js

```diff
@@ -17,11 +40,40 @@ export class AchievementPage extends React.Component {
         });
     }
 
+    // {"id":1,"publishedAt":null,"year":2020,"message":"message"}
     render() {
+        const achievements = this.state.achieveList;
+
+        let sorted = {};
+        achievements.forEach((achievement) => {
+            if (!(achievement.year in sorted)) {
+                sorted[achievement.year] = [];
+            }
+            sorted[achievement.year].push(achievement);
+        });
+
+        let years = Object.keys(sorted);
+        years.sort().reverse();
```

Comment:
I've actually had problems with `Object.keys`, especially in Internet Explorer.
Look around the codebase for examples where we use `import {...} from lodash;` to see how we use the `lodash` library to make our lives easier. You can find documentation here: https://lodash.com/docs/4.17.15

In this case, I'd probably look at the `_.keys` function